Changes in this file:
- do not use -funwind-tables and -latomic on armv7
- run tests with datasize, fds, stacksize and processes at the hard limit

Index: src/Makefile
--- src/Makefile.orig
+++ src/Makefile
@@ -66,7 +66,7 @@ ifneq (,$(filter aarch64 armv,$(uname_M)))
         CFLAGS+=-funwind-tables
 else
 ifneq (,$(findstring armv,$(uname_M)))
-        CFLAGS+=-funwind-tables
+#        CFLAGS+=-funwind-tables
 endif
 endif
 
@@ -97,7 +97,7 @@ DEBUG=-g -ggdb
 
 # Linux ARM32 needs -latomic at linking time
 ifneq (,$(findstring armv,$(uname_M)))
-        FINAL_LIBS+=-latomic
+#        FINAL_LIBS+=-latomic
 endif
 
 ifeq ($(uname_S),SunOS)
@@ -390,7 +390,7 @@ distclean: clean
 .PHONY: distclean
 
 test: $(REDIS_SERVER_NAME) $(REDIS_CHECK_AOF_NAME) $(REDIS_CLI_NAME) $(REDIS_BENCHMARK_NAME)
-	@(cd ..; ./runtest)
+	@(cd ..; ulimit -Sd `ulimit -Hd`; ulimit -Sn `ulimit -Hn`; ulimit -Sp `ulimit -Hp`; ulimit -Ss `ulimit -Hs`; ${TCL_BIN} tests/test_helper.tcl)
 
 test-modules: $(REDIS_SERVER_NAME)
 	@(cd ..; ./runtest-moduleapi)
@@ -433,7 +433,7 @@ src/help.h:
 	@../utils/generate-command-help.rb > help.h
 
 install: all
-	@mkdir -p $(INSTALL_BIN)
+	$(INSTALL_DIR) $(INSTALL_BIN)
 	$(call MAKE_INSTALL,$(REDIS_SERVER_NAME),$(INSTALL_BIN))
 	$(call MAKE_INSTALL,$(REDIS_BENCHMARK_NAME),$(INSTALL_BIN))
 	$(call MAKE_INSTALL,$(REDIS_CLI_NAME),$(INSTALL_BIN))
